pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp

```diff
@@ -0,0 +1,1837 @@
+      return rewriter.notifyMatchFailure(
+          op, "non-const keepdim parameter unsupported");
+
+    return success();
+  }
+};
+
+// This reduction op legalization template handles op variants that reduce in
+// only one explicit dim which is provided as a number (rather than a list), and
+// a keep_dims parameter.
+template <typename AtenOpT, ReductionConvFunc ConversionFuncT>
+class ConvertAtenOneDimReductionOp
+    : public ConvertAtenReductionOp<AtenOpT, ConversionFuncT> {
+  using ConvertAtenReductionOp<AtenOpT,
+                               ConversionFuncT>::ConvertAtenReductionOp;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+  LogicalResult readReduceDimsAndKeepDims(AtenOpT op, OpAdaptor adaptor,
+                                          ConversionPatternRewriter &rewriter,
+                                          ElementsAttr &reduceDimsAttr,
+                                          bool &keepDims) const override {
+    int64_t reduceDim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&reduceDim)))
+      return rewriter.notifyMatchFailure(op,
+                                         "non-const dim parameter unsupported");
+    auto reduceDimsType = RankedTensorType::get({1}, rewriter.getI64Type());
+    reduceDimsAttr = DenseIntElementsAttr::get(reduceDimsType,
+                                               llvm::makeArrayRef({reduceDim}));
+
+    keepDims = false;
+    if (!matchPattern(op.keepdim(), m_TorchConstantBool(&keepDims)))
+      return rewriter.notifyMatchFailure(
+          op, "non-const keepdim parameter unsupported");
+
+    return success();
+  }
+};
+
+// This reduction op legalization template handles op variants that reduce all
+// dims does not keep dims.
+template <typename AtenOpT, ReductionConvFunc ConversionFuncT>
+class ConvertAtenAllDimsReductionOp
+    : public ConvertAtenReductionOp<AtenOpT, ConversionFuncT> {
+public:
+  using ConvertAtenReductionOp<AtenOpT,
+                               ConversionFuncT>::ConvertAtenReductionOp;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+  LogicalResult readReduceDimsAndKeepDims(AtenOpT op, OpAdaptor adaptor,
+                                          ConversionPatternRewriter &rewriter,
+                                          ElementsAttr &reduceDimsAttr,
+                                          bool &keepDims) const override {
+    auto self = adaptor.self();
+    auto selfTy = self.getType().template cast<RankedTensorType>();
+
+    // Select all dims to reduce
+    SmallVector<int64_t, 4> reduceDims;
+    for (int64_t i = 0; i < selfTy.getRank(); i++)
+      reduceDims.push_back(i);
+    int64_t N = selfTy.getRank();
+    auto reduceDimsType = RankedTensorType::get({N}, rewriter.getI64Type());
+    reduceDimsAttr = DenseIntElementsAttr::get(reduceDimsType,
+                                               llvm::makeArrayRef(reduceDims));
+    keepDims = false;
+
+    return success();
+  }
+};
+
+/*
```

Comment:
Better to remove unused code. If we need it later, better to add it at that time.